sdks/java/extensions/sql/src/main/java/org/apache/beam/sdk/extensions/sql/impl/transform/BeamBuiltinAggregations.java

```diff
@@ -495,4 +504,81 @@ public Long extractOutput(Long accumulator) {
       return accumulator;
     }
   }
+  /**
+   * Logical_And function implementation
+   *
+   * <p>Returns the logical AND of all non-NULL expressions. Returns NULL if there are zero input
+   * rows or expression evaluates to NULL for all rows.
+   */
+  public static class LogicalAnd extends CombineFn<Boolean, LogicalAnd.Accum, Boolean> {
+    static class Accum {
+      /** Initially, input is empty. */
+      boolean isEmpty = true;
+      /** true if any null value is seen in the input, null values are to be ignored. */
+      boolean isNull = false;
+      /** logical_and operation result. */
+      boolean logicalAnd = true;
+    }
+
+    @Override
+    public Accum createAccumulator() {
+      return new Accum();
+    }
+
+    @Override
+    public Accum addInput(Accum accum, Boolean input) {
+      /** when accum is empty and it sees null, it remains null */
+      if (accum.isEmpty && input == null) {
+        accum.isNull = true;
+        return accum;
+      }
+      /** when accum is null and it sees null, it remains null */
+      if (accum.isNull && input == null) {
+        accum.isNull = true;
+        return accum;
+      }
+      /** when accum is neither null and nor empty, it remains unchanged */
+      if (!accum.isNull && !accum.isEmpty && input == null) {
```

Comment:
If you've already checked `accum.isEmpty && input == null` and `accum.isNull && input == null`, `!accum.isNull && !accum.isEmpty && input == null` can be simplified to `input == null`.

You can handle all three cases with a lot less code:

```
    if (input == null) {
      if (accum.isEmpty) {
        accum.isEmpty = false;
        accum.isNull = true;
      }
      return accum;
    }
```